remote: nginx: [emerg] host not found in upstream

Having difficulties to understand why "git push dokku master" previously succeeded, failed. No code change happened. Code works well locally.

https://github.com/koajs/todo

---------------------------------------
=====> end koa-todo container output
-----> Running post-deploy
-----> Configuring koa-todo.mydomain.com...(using /var/lib/dokku/plugins/available/nginx-vhosts/templates/nginx.conf.template)
-----> Creating http nginx.conf
-----> Running nginx-pre-reload
       Reloading nginx
remote: nginx: [emerg] host not found in upstream "koa-todo" in /home/dokku/koa-todo/nginx.conf:16
remote: nginx: configuration file /etc/nginx/nginx.conf test failed
---------------------------------------

nginx error log is empty

Any ideas or hints are greatly appreciated

redux-search git:(master) git push dokku master
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
	LANGUAGE = (unset),
	LC_ALL = (unset),
	LC_CTYPE = "UTF-8",
	LANG = "en_US.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to a fallback locale ("en_US.UTF-8").
Counting objects: 509, done.
Delta compression using up to 4 threads.
Compressing objects: 100% (222/222), done.
Writing objects: 100% (509/509), 190.21 KiB | 0 bytes/s, done.
Total 509 (delta 274), reused 508 (delta 274)
-----> Cleaning up...
-----> Building redux-search from herokuish...
-----> Adding BUILD_ENV to build environment...
-----> Node.js app detected

-----> Creating runtime environment

       NPM_CONFIG_LOGLEVEL=error
       NPM_CONFIG_PRODUCTION=true
       NODE_ENV=production
       NODE_MODULES_CACHE=true

-----> Installing binaries
       engines.node (package.json):  unspecified
       engines.npm (package.json):   unspecified (use default)

       Resolving node version (latest stable) via semver.io...
       Downloading and installing node 5.1.1...
       Using default npm version: 3.3.12

-----> Restoring cache
       Loading 2 from cacheDirectories (default):
       - node_modules
       - bower_components (not cached - skipping)

-----> Building dependencies
       Pruning any extraneous modules
       Installing node modules (package.json)

-----> Caching build
       Clearing previous node cache
       Saving 2 cacheDirectories (default):
       - node_modules
       - bower_components (nothing to cache)

-----> Build succeeded!
       +-- node-uuid@1.4.7
       +-- react-highlight-words@0.1.1
       `-- redux@3.0.5

-----> Discovering process types
       Default types for  -> web
-----> Releasing redux-search (dokku/redux-search:latest)...
-----> Deploying redux-search (dokku/redux-search:latest)...
-----> DOKKU_SCALE file found (/home/dokku/redux-search/DOKKU_SCALE)
=====> web=1
-----> Running pre-flight checks
       For more efficient zero downtime deployments, create a file CHECKS.
       See http://progrium.viewdocs.io/dokku/checks-examples.md for examples
       CHECKS file not found in container: Running simple container check...
-----> Waiting for 10 seconds ...
remote: App container failed to start!!
=====> redux-search container output:
       > redux-search@1.0.0 start /app
       > webpack-dev-server --hot --inline --config webpack.config.dev.js
       sh: 1: webpack-dev-server: not found
       npm ERR! Linux 4.2.0-19-generic
       npm ERR! argv "/app/.heroku/node/bin/node" "/app/.heroku/node/bin/npm" "start"
       npm ERR! node v5.1.1
       npm ERR! npm  v3.3.12
       npm ERR! file sh
       npm ERR! code ELIFECYCLE
       npm ERR! errno ENOENT
       npm ERR! syscall spawn
       npm ERR! redux-search@1.0.0 start: `webpack-dev-server --hot --inline --config webpack.config.dev.js`
       npm ERR! spawn ENOENT
       npm ERR!
       npm ERR! Failed at the redux-search@1.0.0 start script 'webpack-dev-server --hot --inline --config webpack.config.dev.js'.
       npm ERR! Make sure you have the latest version of node.js and npm installed.
       npm ERR! If you do, this is most likely a problem with the redux-search package,
       npm ERR! not with npm itself.
       npm ERR! Tell the author that this fails on your system:
       npm ERR!     webpack-dev-server --hot --inline --config webpack.config.dev.js
       npm ERR! You can get their info via:
       npm ERR!     npm owner ls redux-search
       npm ERR! There is likely additional logging output above.
       npm ERR! Please include the following file with any support request:
       npm ERR!     /app/npm-debug.log
=====> end redux-search container output
remote: Error response from daemon: Cannot kill container 3ba3460482d7056b5873f1c1718d1030e97c7d975fb09058b242ee23b66fb70e: notrunning: Container 3ba3460482d7056b5873f1c1718d1030e97c7d975fb09058b242ee23b66fb70e is not running
remote: Error: failed to kill containers: [3ba3460482d7056b5873f1c1718d1030e97c7d975fb09058b242ee23b66fb70e]
To dokku@something.com:redux-search
 ! [remote rejected] master -> master (pre-receive hook declined)
error: failed to push some refs to 'dokku@something.com:redux-search'
➜  redux-search git:(master)

@josegonzalez  

Thnx! Destroying redux-search helped. 

I did few changes in todo apps with following, commit&push. It works well.
https://github.com/koajs/todo

I've also tried to re-deployed redux-search after, and it corrupted "git push" for todo app again.
https://github.com/koajs/todo

It seems that somehow redux-search corrupts my dokku setup. redux-search "npm start" looks bad to me - running webpack server is weird from my perspective. 

    "start": "webpack-dev-server --hot --inline --config webpack.config.dev.js",
https://github.com/treasure-data/redux-search/blob/master/package.json

So I guess it cause all the mess:

-----> Running pre-flight checks
       For more efficient zero downtime deployments, create a file CHECKS.
       See http://progrium.viewdocs.io/dokku/checks-examples.md for examples
       CHECKS file not found in container: Running simple container check...
-----> Waiting for 10 seconds ...
remote: App container failed to start!!
=====> redux-search container output:
       > redux-search@1.0.0 start /app
       > webpack-dev-server --hot --inline --config webpack.config.dev.js
       sh: 1: webpack-dev-server: not found
       npm ERR! Linux 4.2.0-19-generic
       npm ERR! argv "/app/.heroku/node/bin/node" "/app/.heroku/node/bin/npm" "start"
       npm ERR! node v5.1.1
       npm ERR! npm  v3.3.12
       npm ERR! file sh
       npm ERR! code ELIFECYCLE
       npm ERR! errno ENOENT
       npm ERR! syscall spawn
       npm ERR! redux-search@1.0.0 start: `webpack-dev-server --hot --inline --config webpack.config.dev.js`
       npm ERR! spawn ENOENT
       npm ERR!
       npm ERR! Failed at the redux-search@1.0.0 start script 'webpack-dev-server --hot --inline --config webpack.config.dev.js'.
       npm ERR! Make sure you have the latest version of node.js and npm installed.
       npm ERR! If you do, this is most likely a problem with the redux-search package,
       npm ERR! not with npm itself.
       npm ERR! Tell the author that this fails on your system:
       npm ERR!     webpack-dev-server --hot --inline --config webpack.config.dev.js
       npm ERR! You can get their info via:
       npm ERR!     npm owner ls redux-search
       npm ERR! There is likely additional logging output above.
       npm ERR! Please include the following file with any support request:
       npm ERR!     /app/npm-debug.log
=====> end redux-search container output
remote: Error response from daemon: Cannot kill container 3ba3460482d7056b5873f1c1718d1030e97c7d975fb09058b242ee23b66fb70e: notrunning: Container 3ba3460482d7056b5873f1c1718d1030e97c7d975fb09058b242ee23b66fb70e is not running
remote: Error: failed to kill containers: [3ba3460482d7056b5873f1c1718d1030e97c7d975fb09058b242ee23b66fb70e]

--- a/plugins/nginx-vhosts/functions
+++ b/plugins/nginx-vhosts/functions
@@ -235,8 +235,13 @@ EOF
       echo "}" >> $NGINX_CONF
     fi
 
-    dokku_log_info1 "Creating $SCHEME nginx.conf"
-    mv $NGINX_CONF "$DOKKU_ROOT/$APP/nginx.conf"
+    if is_deployed "$APP"; then
+      dokku_log_info1 "Creating $SCHEME nginx.conf"
+      mv $NGINX_CONF "$DOKKU_ROOT/$APP/nginx.conf"
+    else
+      dokku_log_info1 "App $APP has not been deployed. Skipping nginx config creation"
+      rm -f $NGINX_CONF
+    fi
 
     if is_deployed "$APP"; then
       dokku_log_info1 "Running nginx-pre-reload"

